### DISCUSSION TITLE
Memory issue on encrypting large files

### DISCUSSION THREAD

**Discussion by tobiasjungnickel** [krzyzanowskim/ObjectivePGP#211] [General]
- ObjectivePGP version: 0.12.0
- Platform (iOS/macOS): iOS 11.4 / iPhone 7
- Installation method (CocoaPods/Carthage/...): CocoaPods

# Description

When encrypting a "large" file (tested with 100M) our App's ShareExtension target using ObjPGP terminated due to memory issue. No error is thrown, only console output:
`Message from debugger: Terminated due to memory issue`
`Program ended with exit code: 0`

# Steps to reproduce

Sample code to reproduce the issue:

```objective-c

class func encrypt(_ data: Data, addSignature sign: Bool, using keys: [Key], passphraseForKey passphraseBlock: ((Key) -> String?)? = nil) throws -> Data

```

**Comment by krzyzanowskim** [comment]
A known issue #102
